### ISSUE TITLE
`misc.but No paragraph should start with a 'But'`: checks starts of sentences too, not just paragraphs (rst/TeX)

### ISSUE THREAD

**Issue by sanjayankur31** [amperser/proselint#1376]
The warning is meant to be shown when "but" is at the start of paragraphs. However, in its current implementation, it seens ti check for "but" at the beginning of each new line in the body of the text.  This is incorrect for formats that normalises white spaces and allow semantic line breaks.  They can have "but" at the start of a new line within a paragraph (at the start of or within a sentence).

In rst/markdown/TeX, a new paragraph must be preceded by an empty line.

So:

```
This is a sentence
but this is not the beginning of a new paragraph.
The warning should not be shown here.

But, it should be shown here.
```

The warning should appear at the second "but", but not at the first.

**Comment by Nytelife26** [comment]
This is meant to be addressed in the code, by using a case-sensitive match. If you're getting false positives from it anyway, that indicates a problem with the `ignore_case` switch.

I'll try to verify this later, although the check has been updated in #1371, which may have inadvertently fixed this issue. I'll get back to you either way.

**Comment by sanjayankur31** [comment]
Thanks for looking into this @Nytelife26 . Sorry, my issues description wasn't clear. I've updated it now.

```
A sentence
but continues in the next line
```

does not generate a warning. So this case is fine.

However, with a capital "But" at the start of a sentence, but not a new paragraph, it still does:

```
A sentence.
But at the start of new sentence not a new paragraph in rst/markdown/TeX.
```

In formats like rst/markdown/TeX, this isn't  at the start of a new paragraph. A new paragraph needs to be preceded by an empty line in these formats:


```
A sentence or paragraph.

But at the start of a new paragraph, because there's an empty line before this sentence.
```

So, I think the check needs to be on the lines of: `"But" (capital b) at the start of a sentence that is preceded by an empty line`.

**Comment by Nytelife26** [comment]
it was perfectly clear, looking back on it. i was tired when i read your issue and wrote the reply, so my apologies for that.

this should be relatively trivial to fix, and i'll add a test for it while i'm at it.
